alica_engine/src/engine/modelmanagement/factories/BlackboardBlueprintFactory.cpp

```diff
@@ -12,10 +12,7 @@ std::unique_ptr<BlackboardBlueprint> BlackboardBlueprintFactory::create(const YA
     for (const auto& entry : node) {
         auto key = getValue<std::string>(entry, Strings::key);
         auto type = getValue<std::string>(entry, Strings::stateType);
-        std::optional<std::string> defaultValue;
-        if (!entry[Strings::defaultValue].IsNull()) {
-            defaultValue = getValue<std::string>(entry, Strings::defaultValue);
-        }
+        std::optional<std::string> defaultValue; // deprecated
```

Comment:
I don't know if I got this. `defaultValue` is not being assigned, the old test to check if the yaml file had a defaultValue value was removed, and always something with a value equals std::nullopt is being passed to `blueprint->addKey`.

Shouldn't it be `blueprint->addKey(key, type, std::nullopt);` and we remove line 15?

Reply:
Yeah that's fine too, a follow up will have to clean up the function signature of addKey either way to remove the default value

Reply:
Was trying to just do minimal changes to unbreak stuff but I went and properly removed the field

Reply:
Actually properly removing seems quite difficult as there are a lot of tests.  Since the purpose of this PR is mainly to unblock use of the plan designer, I will leave it as is, just deprecating the function signature which will give warnings when compiling the tests.